statistics/fmsketch.go

```diff
@@ -122,3 +122,33 @@ func FMSketchFromProto(protoSketch *tipb.FMSketch) *FMSketch {
 	}
 	return sketch
 }
+
+// EncodeFMSketch encodes the given FMSketch to byte slice.
+func EncodeFMSketch(c *FMSketch) ([]byte, error) {
+	if c == nil {
+		return nil, nil
+	}
+	p := FMSketchToProto(c)
+	protoData, err := p.Marshal()
+	return protoData, err
+}
+
+// DecodeFMSketch decode a FMSketch from the given byte slice.
+func DecodeFMSketch(data []byte) (*FMSketch, error) {
+	if data == nil {
+		return nil, nil
+	}
+	p := &tipb.FMSketch{}
+	err := p.Unmarshal(data)
+	if err != nil {
+		return nil, errors.Trace(err)
+	}
+	fm := FMSketchFromProto(p)
+	return fm, nil
+}
+
+// MemoryUsage returns the total memory usage of a FMSketch.
+func (c *FMSketch) MemoryUsage() (sum int64) {
+	// TODO: sum = ?
```

Comment:
Do you have any suggestions for the fmSketch calculation formula for memory usage? @qw4990 @rebelice @xuyifangreeneyes